core/server/themes/index.js

```diff
@@ -22,8 +23,17 @@ module.exports = {
         // Just read the active theme for now
         return themeLoader
             .loadOneTheme(activeThemeName)
-            .then(function activeThemeHasLoaded() {
-                debug('Activating theme (method A on boot)', activeThemeName);
+            .then(function activeThemeHasLoaded(theme) {
+                // Validate
+                return validate
```

Comment:
Yes this must always happen on bootstrap. Why - because we HAVE to get certain information about a theme in order for the theme to work.

We use gscan here because gscan is designed to do this work for us. It needs improvement, but its purpose is to read a theme and determine all sorts of details about the structure.

If we don't use gscan, then we have to re-implement the reading of templates, partials, etc outside of gscan. We already use gscan for uploads and activations, it makes no sense to me to implement the same logic again in Ghost and do the same work twice?

Yes, this will slow down the bootstrap - but don't forget custom theme templates is currently unimplemented. I have to reimplement it before we ship!

Perhaps there are ways to make gscan faster / more efficient?